### PR TITLE
Verify client certificate through REST API

### LINKED ISSUE #195: When changing credentials old cache prevents connection
The old persistency of a device might prevent device connection when the credential secret has been changed.

To replicate:
- Connect a device to MQTT using a credential secret.
- Disconnect and destroy the device.
- While disconnected, wipe the device credentials in the Astarte instance. Then perform the manual registration once more obtaining a new credential secret.
- Do not modify the device persistency folder. Attempt to reconnect the device using the new credential secret.

The device will not return an error but cannot connect to Astarte.

### PR DESCRIPTION
**Pull request by sorru94**
Fixes #195

### CODE REVIEW

File: astarte/device/device_mqtt.py [hunk 1]
```diff
@@ -458,7 +466,14 @@ def __on_disconnect(self, _client, _userdata, rc):
             self.__mqtt_client.loop_stop()
         # Else check certificate expiration and try reconnection
         # TODO: check for MQTT_ERR_TLS when Paho correctly returns it
```
Comment (harlem88) [inline astarte/device/device_mqtt.py:468]: Is it an old comment? Perhaps you can remove it

**Review by harlem88 — CHANGES_REQUESTED** [review]

File: astarte/device/device_mqtt.py (hunk 1 above)
Comment (sorru94) [reply astarte/device/device_mqtt.py:468]: Yes, looking at the sources of paho for Python it appears that `MQTT_ERR_TLS` is not used.

**Review by harlem88 — COMMENTED** [review]
Add changes to the Changelog

File: astarte/device/crypto.py [hunk 2]
```diff
@@ -137,13 +181,12 @@ def certificate_is_valid(crypto_store_dir: str) -> bool:
...
-        data = file.read()
-    if data:
-        try:
-            certificate = x509.load_pem_x509_certificate(data.encode("ascii"))
-        except ValueError:
-            return False
-        return certificate.not_valid_before < datetime.utcnow() < certificate.not_valid_after
+        cert_pem = file.read()
+        if cert_pem:
+            return pairing_handler.verify_device_certificate(
```
[the remark this answers is not in the document]
Comment (sorru94) [reply astarte/device/crypto.py:189]: I don't think the overhead is justified by saving an HTTP request. So I would not do both.

**Review by lucaato — APPROVED** [review]

**Review by harlem88 — APPROVED** [review]

**Pull request merged**